scipy/stats/tests/test_distributions.py

```diff
@@ -3990,6 +3990,98 @@ def test_burr_nan_mean_var_9544(self):
         assert_(np.isfinite(e4))
 
 
+class TestStudentizedRange(object):
+    # Tables are arranged with the first number in each subarray
+    # indicating v, with the second value ([1]) at k=2. k increases by
+    # 6 for each new value to test a range of values.
+    # The format is [v, k=2, k=8, k=14, k=20]
+
+    alpha05 = [[1, 17.97, 45.40, 54.33, 59.56],
+               [3, 4.501, 8.853, 10.35, 11.24],
+               [10, 3.151, 5.305, 6.028, 6.467],
+               [20, 2.950, 4.768, 5.357, 5.714]]
+
+    alpha01 = [[1, 90.03, 227.2, 271.8, 298.0],
+               [3, 8.261, 15.64, 18.22, 19.77],
+               [10, 4.482, 6.875, 7.712, 8.226],
+               [20, 4.024, 5.839, 6.450, 6.823]]
+
+    def test_cdf_against_tables(self):
+        def _test_set_cdf(set, significance):
+            v = set[0]
+            for k_ind in range(3):
+                k = 2 + k_ind * 6
+                q = set[1 + k_ind]
+                res_p = stats.studentized_range.cdf(q, k, v)
+
+                # In assertion, only expect as many sig-figs as we have input
+                assert_allclose(res_p, significance, rtol=1e-4)
+
+        for test_set in self.alpha05:
+            _test_set_cdf(test_set, 0.95)
+
+        for test_set in self.alpha01:
+            _test_set_cdf(test_set, 0.99)
+        return
+
+    @pytest.mark.slow
+    def test_ppf_against_tables(self):
+        def _test_set_ppf(set, significance):
+            v = set[0]
+            for k_ind in range(3):
+                k = 2 + k_ind * 6
+                q = set[1 + k_ind]
+                res_q = stats.studentized_range.ppf(significance, k, v)
+                # Accuracy is somthing to be improved on.
+                assert_allclose(res_q, q, rtol=5e-4)
+
+        for test_set in self.alpha05:
+            _test_set_ppf(test_set, 0.95)
+
+        for test_set in self.alpha01:
+            _test_set_ppf(test_set, 0.99)
+        return
```

Comment:
```suggestion
 # Critical Q values were selected from this table:
    # https://www.jstor.org/stable/2237810
    # For alpha=.05 and .01, and for each value of v=[1, 3, 10, 20],
    # a Q was picked from each table for k=[2, 8, 14, 20].
    ks = [2, 8, 14, 20]
    vs = [1, 3, 10, 20]

    alpha05 = [17.97, 45.40, 54.33, 59.56,
               4.501, 8.853, 10.35, 11.24,
               3.151, 5.305, 6.028, 6.467,
               2.950, 4.768, 5.357, 5.714]

    alpha01 = [90.03, 227.2, 271.8, 298.0,
               8.261, 15.64, 18.22, 19.77,
               4.482, 6.875, 7.712, 8.226,
               4.024, 5.839, 6.450, 6.823]

    ks_d = np.ravel(list((repeat(ks, 4))))
    vs_d = np.ravel(list(zip(*(repeat(vs, 4)))))
    p95 = [.95] * 16
    p99 = [.99] * 16

    sig_k_v_q = (list(zip(p95, ks_d, vs_d, alpha05))
                 + list(zip(p99, ks_d, vs_d, alpha01)))

    @pytest.mark.parametrize("significance, k, v,q", sig_k_v_q)
    def test_cdf_against_tables(self, significance, k, v, q):
        res_p = stats.studentized_range.cdf(q, k, v)
        # some table values are only accurate to the second decimal place.
        atol = 1e-3 if q < 10 else 1e-2 if q < 100 else 1e-1
        assert_allclose(res_p, significance, atol=atol)

    @pytest.mark.parametrize("significance,k,v,q", sig_k_v_q)
    def test_ppf_against_tables(self, significance, k, v, q):
        res_q = stats.studentized_range.ppf(significance, k, v)
        # some table values are only accurate to the second decimal place.
        atol = 1e-3 if q < 10 else 1e-2 if q < 100 else 1e-1
        assert_allclose(res_q, q, atol=atol)

```

Reply:
Parametrized these. I also cross checked the q values against https://www.jstor.org/stable/2237810, which seems to be an original source on the tables.